main/test/flix/Test.Exp.ArraySlice.flix

```diff
@@ -3,203 +3,203 @@ namespace Test/Exp/ArraySlice {
     @test
     def testArraySlice01(): Unit = region r {
         let x = [1, 2, 3] @ r;
-        let _ = x[0..2];
+        discard x[0..2];
         ()
     }
 
     @test
     def testArraySlice02(): Unit = region r {
         let x = [1, 2, 3, 4] @ r;
-        let _ = x[2..3];
+        discard x[2..3];
         ()
     }
 
     @test
     def testArraySlice03(): Unit = region r {
         let x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20] @ r;
-        let _ = x[10..15];
+        discard x[10..15];
         ()
     }
 
     @test
     def testArraySlice04(): Unit = region r {
         let x = [1, 2] @ r;
-        let _ = x[1..2];
+        discard x[1..2];
         ()
     }
 
     @test
     def testArraySlice05(): Unit = region r {
         let x = [true, false] @ r;
-        let _ = x[0..1];
+        discard x[0..1];
         ()
     }
 
     @test
     def testArraySlice06(): Unit = region r {
         let x = [false, true, false] @ r;
-        let _ = x[2..3];
+        discard x[2..3];
         ()
     }
 
     @test
     def testArraySlice07(): Unit = region r {
         let x = [false, false, false, false, true] @ r;
-        let _ = x[1..3];
+        discard x[1..3];
```

Comment:
I am surprised? I would think that slicing a mutable array has an effect?

Reply:
It does have an effect, but it is non-unit, so you have to discard it. discarding something pure would be an error.